File: android/src/main/java/com/websquare/plugins/zendesksupport/ZendeskSupport.kt
package com.websquare.plugins.zendesksupport

import android.content.Context
import android.util.Log
import zendesk.android.Zendesk
import zendesk.messaging.android.DefaultMessagingFactory

class ZendeskSupport {
    fun initialize(context: Context?, url: String?, appId: String?, clientId: String?, androidChatId: String?, debugLog: Boolean) {
        if (debugLog) Log.i("Initialized", "Done")

        Zendesk.initialize(
                context = context!!,
                channelKey = androidChatId!!,
                successCallback = { zendesk ->
                    Log.i("IntegrationApplication", zendesk.toString())
                    Log.i("IntegrationApplication", "Initialization successful")
                },
                failureCallback = { error ->
                    // Tracking the cause of exceptions in your crash reporting dashboard will help to triage any unexpected failures in production
                    Log.e("IntegrationApplication", "Initialization failed", error)
                },
                messagingFactory = DefaultMessagingFactory()
        )
    }

    fun setAnonymousIdentity(name: String?, email: String?) {
        Log.i("Warning", "No anonymous identity")
    }

    fun setIdentity(token: String?) {
        Zendesk.instance.loginUser(jwt = token!!,
                successCallback = { user ->
                    Log.i("User", user.toString())
                },
                failureCallback = { error ->
                    Log.i("Error", error.toString())
                }
        )
    }

    fun openChat(context: Context) {
        Zendesk.instance.messaging.showMessaging(context)
    }
}

File: ios/Plugin/ZendeskSupport.swift
import Foundation
import ZendeskSDKMessaging
import ZendeskSDK

@objc public class ZendeskSupport: NSObject {
    @objc public func initialize(_ appId: String,_ clientId: String,_ zendeskUrl: String,_ messagingId: String) throws {
        Zendesk.initialize(withChannelKey: messagingId,
                           messagingFactory: DefaultMessagingFactory()) { result in
                if case let .failure(error) = result {
                    print("Messaging did not initialize.\nError: \(error.localizedDescription)")
                }
            }
    }

    @objc public func setAnonymousIdentity(_ name: String,_ email: String) {
        print("No anonimous users")
    }

    @objc public func setIdentity(_ token: String) {
        Zendesk.instance?.loginUser(with: token) { result in
            switch result {
            case .success(let user):
                print("User: \(user)");
            case .failure(let error):
                print(error);
            }
        }
    }

    @objc public func openChat(_ viewCtrl: UIViewController?) {
        DispatchQueue.main.async {
            guard let viewController = Zendesk.instance?.messaging?.messagingViewController() else { return }
            let navController = UINavigationController(rootViewController: viewController)
            viewCtrl?.present(navController, animated: true)
        }
    }
}
